File: java/VendingMachine/src/com/ravendarque/items/Item.java
package com.ravendarque.items;

public class Item {

    private final String displayName;
    private double price;

    public Item(String displayName, double price) {

        validateDisplayName(displayName);
        validatePrice(price);

        this.displayName = displayName;
        this.price = price;
    }

    public String getDisplayName() {

        return displayName;
    }

    public double getPrice() {

        return price;
    }

    private void validateDisplayName(String displayName) {

        if (displayName == null)
            throw new IllegalArgumentException("Display name cannot be null");

        if (displayName.equals(""))
            throw new IllegalArgumentException("Display name cannot be empty");
    }

    private void validatePrice(double price) {

        if (price < 0)
            throw new IllegalArgumentException("Price cannot be negative");

    }
}

File: C++/VendingMachine/VendingMachine/Item.cpp
#include "pch.h"
#include "Item.h"
#include <utility>

Item::Item(std::string displayName, const double price): DisplayName(std::move(displayName)), Price(price)
{
}

std::string Item::GetDisplayName() const
{
	return DisplayName;
}

double Item::GetPrice() const
{
	return Price;
}

bool Item::operator==(Item& rhs) const
{
	return (GetDisplayName() == rhs.GetDisplayName() &&
		GetPrice() == rhs.GetPrice());
}

bool Item::operator==(const Item& rhs) const
{
	return (GetDisplayName() == rhs.GetDisplayName() &&
		GetPrice() == rhs.GetPrice());
}
